### PR TITLE
feat: Initial commit

### PR DESCRIPTION
**Pull request by lfvjimisola**
This PR will be the initial commit of the reqstool-client released as open source. It contains the pre-existing source code as well as newly created GitHub workflows.

### CODE REVIEW

File: tests/unit/reqstool/commands/status/test_statistics_generator.py [hunk 1]
```diff
@@ -0,0 +1,463 @@
...
+            nr_of_total_tests=4,
+            nr_of_completed_requirements=2,
+            nr_of_total_requirements=4,
+            nr_of_reqs_with_implementation=2,
+        ),
+    )
+    assert result == expected
+
+
+# @pytest.mark.skip(reason="Might be testdata error. Need to investigate")
```
Comment (lfvbarcus) [inline tests/unit/reqstool/commands/status/test_statistics_generator.py:119]: This out commented line should be removed

File: tests/unit/reqstool/model_generators/test_combined_raw_datasets_generator.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    semantic_validator = SemanticValidator(validation_error_holder=ValidationErrorHolder())
+    combined_raw_datasets_generator.CombinedRawDatasetsGenerator(
+        initial_location=LocalLocation(
+            path=local_testdata_resources_rootdir_w_path("test_basic/with_requirements_config/ms-101")
+        ),
+        semantic_validator=semantic_validator,
+    )
+
+
+@pytest.mark.skip(reason="Test should target combined indexed data set instead of CombinedRawDataset")
```
Comment (lfvbarcus) [inline tests/unit/reqstool/model_generators/test_combined_raw_datasets_generator.py:29]: This seems like the exactly the same test as the one below it, but with different testdata. Perhaps remove this and focus on acually validate some of the response from the test below instead? We could create a ticket for that as well as write the test marked as TODO in the test_testdata_model_generator?

File: tests/unit/reqstool/model_generators/test_testdata_model_generator.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+
+@pytest.mark.parametrize("method_name", unit_method_names)
+def test_unit_method_identifier_regex(method_name):
+    unit_match = re.match(TestDataModelGenerator.UNIT_METHOD_IDENTIFIER_REGEX, method_name)
+    assert unit_match is not None
+    assert unit_match.group(1) == "testFlightIdAircraftId"
+
+
+@pytest.mark.skip(reason="Awaiting testdata update")
```
Comment (lfvbarcus) [inline tests/unit/reqstool/model_generators/test_testdata_model_generator.py:35]: This should be completed. Let's open a separate issue for that?

Comment (lfvjimisola) [reply tests/unit/reqstool/model_generators/test_testdata_model_generator.py:35]: Ok. Once code and CI/CD has been migrated here we'll also move over existing tickets.

File: docs/modules/ROOT/pages/quickstart.adoc [hunk 4]
```diff
@@ -0,0 +1,7 @@
+
+## Quickstart
+
+. Create a requirements.yml file (see [Link examples](../examples/) or [JSON Schema](../schemas/))
+. Add annotations to your code
+. Add parsing of Reqstool annotations to your build (currently, only Java is supported with `reqstool-annotations-java`)
```
Comment (lfvbarcus) [inline docs/modules/ROOT/pages/quickstart.adoc:6]: Nitpicking, but we actually support Python as well :)

**Review by lfvbarcus — DISMISSED** [review]
I think this looks good! One nitpick comment in the docs.

File: .github/workflows/publish_development.yml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    # Specifying a GitHub environment is optional, but strongly encouraged
+    environment:
+      name: development
+      url: https://test.pypi.org/p/reqstool-python-decorators
+    permissions:
+      id-token: write # IMPORTANT: this permission is mandatory for trusted publishing
+    steps:
+      # Download artifacts from the build job
+      - name: Download Artifacts
+        uses: actions/download-artifact@v3
```
Comment (lfvbarcus) [inline .github/workflows/publish_development.yml:30]: Maybe update this to v4?

File: .github/workflows/lint.yml [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  workflow_call:
+
+jobs:
+  linting:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out source repository
+        uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v4
```
Comment (lfvbarcus) [inline .github/workflows/lint.yml:12]: Maybe update this to v5?

File: .github/workflows/build.yml [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+          python-version: "3.10"
+      - name: Install dependencies
+        run: pip install hatch
+      - name: Run unit and integrations tests
+        run: hatch run test:pytest --cov=reqstool-python-decorators --cov-report=xml --cov-report=html 
+      - name: Build project
+        run: hatch build
+      # Upload artifacts for later use
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v3
```
Comment (lfvbarcus) [inline .github/workflows/build.yml:30]: Maybe update this to v4?

**Review by lfvbarcus — DISMISSED** [review]

**Review by lfvbarcus — APPROVED** [review]

**Pull request merged**